src/code_oscar.jl

```diff
@@ -0,0 +1,566 @@
+
+    collector_cones = []
+
+    # We need not consider sets of smaller size because of the rank condition.
+    for k in projected_dimension:nr_variables
+        set = GAP.Globals.Combinations(GAP.GapObj(1:nr_variables), k)
+        orbs = GAP.Globals.Orbits(G.X, set, GAP.Globals.OnSets)
+        reps = [orbs[i][1] for i in 1:length(orbs)]
+        reps_julia = [Vector{Int64}(i) for i in reps]
+
+        for i in reps_julia
+            current_mat = Q[i,:]
+            if Nemo.rank(current_mat) == projected_dimension &&
+               is_monomial_free(I, setdiff(1:nr_variables, i))
+                cone = Polymake.polytope.Cone(INPUT_RAYS = current_mat)
+                if ! any(j -> Polymake.polytope.equal_polyhedra(j, cone),
+                      collector_cones)
+                    push!(collector_cones, cone)
+                end
+            end
+        end
+    end
+
+    return collector_cones
+end
+#T what if some projections lie in the same orbit?
+#T later we expand the orbits, do we want to check this here?
+
+@doc Markdown.doc"""
+    action_on_target(Q::Array{Int,2}, G::Oscar.GAPGroup)
+
+Let `Q` be an $n \times m$ Q-matrix, and `G` be a GAP permutation group
+on $n$ points that describes an action on the rows of `Q`.
+The function returns the group homomorphism $\rho$ from `G`
+to its induced matrix action on the $m$-dimensional space over the Rationals.
+
+# Examples
+
+```jldoctest
+julia> Q = [
+        1  1   0   0   0 ;
+        1  0   1   1   0 ;
+        1  0   1   0   1 ;
+        1  0   0   1   1 ;
+        0  1   0   0  -1 ;
+        0  1   0  -1   0 ;
+        0  1  -1   0   0 ;
+        0  0   1   0   0 ;
+        0  0   0   1   0 ;
+        0  0   0   0   1 ];
+
+julia> n = size(Q, 1)
+10
+
+julia> perms_list = [[1,3,2,4,6,5,7,8,10,9], [5,7,1,6,9,2,8,4,10,3]];
+
+julia> sym10 = Oscar.symmetric_group(n);
+
+julia> permgens = [sym10(x) for x in perms_list];
+
+julia> G, emb = sub(permgens...);
+
+julia> GITFans.action_on_target(Q, G)
+Group homomorphism from 
+Group([ (2,3)(5,6)(9,10), (1,5,9,10,3)(2,7,8,4,6) ])
+to
+GL(5,Integers)
+
+
+```
+
+Note that $\rho(\pi)$, for $\pi \in $`G`, satisfies
+`Q`$^\pi$ = `Q` * $\rho(\pi^{-1})$, where `Q`$^\pi$ is the matrix obtained
+from `Q` by permuting its rows by $\pi$.
+
+Let $Q\{J\}$ denote the matrix whose rows are those rows of $Q$ indexed by
+the subset $J$ of $\{ 1, 2, \ldots, n \}$.
+The cone defined by $Q{J}$ is mapped to
+$Q{J} \cdot \rho(\pi^{-1}) = (Q \cdot \rho(\pi^{-1}))\{J\} = Q^\pi\{J\}$,
+which is equal to $Q\{J^{\pi^{-1}}\}$.
+"""
+function action_on_target(Q::Array{Int,2}, G::Oscar.GAPGroup)
+
+    # For each permutation generator,
+    # solve the linear equation system.
+    m = size(Q, 1)
+    n = size(Q, 2)
+# FIXME: GAP knows no GL(n, \Q), we take GL(n, \Z) for the moment.
+    matZ = Oscar.matrix(Oscar.ZZ, Q)
+    genimgs = []
+    permgens = Oscar.gens(G)
+    for ppi in permgens
+      imgs = [ppi(i) for i in 1:m]
```

Comment:
This is essentially what GAP's `ListPerm` does, I think. We already have a wrapper for GAP's `PermList` and absolutely should have one for `ListPerm` in Oscar; this could be just a constructor `Vector(perm)` (assuming `m == deg(perm)` )

Reply:
There is already `listperm` in `GAPGroups.jl`. I think `Vector(perm)` could replace it, and then the following would be appropriate.
```
Base.Vector{T}(x::PermGroupElem, n::Int = x.parent.deg) where {T} = T[x(i) for i in 1:n]
```
(I will add this to a pull request for Oscar.)